### PR TITLE
Add needs field for batch action outputs

### PR DESCRIPTION
**Pull request by romasku**

### CODE REVIEW

File: neuro_flow/batch_runner.py [hunk 1]
```diff
@@ -150,7 +150,7 @@ async def _collect_configs_for_task(
             if isinstance(task, ast.BaseActionCall):
                 action_name = await task.action.eval(EMPTY_ROOT)
                 action_path = self._parse_action_name(action_name)
-                result += [LocalPath(action_name)]
+                result += [LocalPath(action_path)]
```
Comment (asvetlov) [inline neuro_flow/batch_runner.py:153]: pls merge the master branch; this line should go from the review page

File: neuro_flow/context.py [hunk 2]
```diff
@@ -1223,6 +1223,15 @@ async def create(  # type:ignore[override]
...
         return cast(_CtxT, replace(ctx, _prefix=prefix, _prep_tasks=prep_tasks))
 
+    async def get_output_needs(self) -> AbstractSet[FullID]:
+        assert isinstance(self._ast, ast.BatchAction)
+        if self._ast.outputs and self._ast.outputs.needs is not None:
+            return {
+                self._prefix + (await need.eval(self),)
+                for need in self._ast.outputs.needs
+            }
+        return self.graph.keys()
```
Comment (asvetlov) [inline neuro_flow/context.py:1233]: I doubt if we need this branch.
Enforcing the `needs` attribute for outputs looks like a sane idea.
What do you think?

Comment (romasku) [reply neuro_flow/context.py:1240]: Both options are OK for me, I have no preference here. On other same, "not enforced -> enforced" is breaking change, so it's better to make it enforced now, and change this if devops tell us to do so.

Comment (asvetlov) [reply neuro_flow/context.py:1240]: Agree. 
Please select the stricter option, we can relax it later on the demand.

Comment (romasku) [reply neuro_flow/context.py:1240]: Done

Comment (asvetlov) [reply neuro_flow/context.py:1240]: I believe the function should return an empty list if no outputs were specified in action's YAML

File: neuro_flow/ast.py [hunk 3]
```diff
@@ -242,9 +242,17 @@ class LiveAction(BaseAction):
     job: Job
 
 
+@dataclass(frozen=True)
+class BatchActionOutputs(Base):
+    # AST class is slightly different from YAML representation,
+    # in YAML `values` mapping is embedded into the outputs itself.
+    needs: Sequence[IdExpr]
+    values: Optional[Mapping[str, Output]] = field(metadata={"allow_none": True})
```
Comment (asvetlov) [inline neuro_flow/ast.py:250]: What is the reason for `values=None`?

Comment (romasku) [reply neuro_flow/ast.py:250]: To be able to do something like this:
```
outputs:
   needs: [final_task] # action should fail if this fails
tasks:
   ...
```

File: neuro_flow/context.py (hunk 2 above)
Comment (romasku) [reply neuro_flow/context.py]: OK, but this will make all actions without outputs never fail.

File: neuro_flow/ast.py (hunk 3 above)
Comment (asvetlov) [reply neuro_flow/ast.py:250]: ok

**Review by asvetlov — APPROVED** [review]
LGTM

**Pull request merged**